### PR TITLE
Add ECDSA s order check

### PR DESCRIPTION
**Pull request by ArnSch**
## Summary

In two flows (`confidentialApprove` and `approveProof`) our replay protection was not strict enough, and flipping the order of `s` and `v` in a provided signature will provide a valid, but different signature anyone can replay once. This is only problematic if a user approves an address, and then revokes that approval. In that scenario, the initially approved address could re-approve themselves once by replaying a signature, assuming the approved notes have not been spent or moved.

There has been no instances of the described flow on mainnet.

This PR fixes this issue, by adding a requirement that `s` be of lower order. No assets deployed after 1st of March 2020 are vulnerable. This PR also adds a patch to our client side libraries and SDK, which will now submit two signatures for every approval and every revoke for assets deployed prior to the 1st of March 2020, making this replay attack un-actionable.

## Types of changes

* Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/extension/src/config/supportsPermitSignature.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export default {
-    '0x6B175474E89094C44Da98b954EedeAC495271d0F': true,
+    // '0x6B175474E89094C44Da98b954EedeAC495271d0F': true,
```
Comment (joeandrews) [inline packages/extension/src/config/supportsPermitSignature.js:2]: We should probably re-enable this I think it got disabled at the hackathon as we hadn't upgraded

Comment (joeandrews) [reply packages/extension/src/config/supportsPermitSignature.js:2]: This is DAI

**Review by joeandrews — COMMENTED** [review]
Just one tiny note about the enabling Dai/Permit in the sdk, we can do that later though.

**Comment by ArnSch** [comment]
:tada: This PR is included in version 0.18.0 :tada:

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

**Comment by ArnSch** [comment]
:tada: This PR is included in version 1.19.0 :tada:

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

**Comment by ArnSch** [comment]
:tada: This PR is included in version 1.21.0 :tada:

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

**Comment by ArnSch** [comment]
:tada: This PR is included in version 1.7.0 :tada:

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

**Pull request merged**